provider/data_source_wireguard_config_document.go

```diff
@@ -356,6 +365,11 @@ func dataSourceWireguardConfigDocumentRead(d *schema.ResourceData, m interface{}
 				peerCfg.PersistentKeepalive = &ka
 			}
 
+			if v := peer["description"]; v != "" {
+				desc := v.(string)
+				peerCfg.Description = &desc
```

Comment:
Let's either strip newlines here, or else split on them and put each on a new comment line in the template.